src/Umbraco.Web.UI.Client/src/common/directives/components/events/events.directive.js

```diff
@@ -101,14 +101,14 @@ angular.module('umbraco.directives')
         var eventBindings = [];
 
         function oneTimeClick(event) {
-                var el = event.target.nodeName;
-
-                //ignore link and button clicks
-                var els = ["INPUT","A","BUTTON"];
-                if(els.indexOf(el) >= 0){return;}
+                // ignore clicks on button groups toggles (i.e. the save and publish button)
```

Comment:
I know that there was hard coded things and DOM checks in here before but just wondering if there's a way to make this a 'real' directive that isn't relying on hard coded classes, DOM checks, etc... ?

I'm sure this will be fine to test/merge for now but would be much nicer if we can think of a better way to do these things. /cc @nielslyngsoe ?

Reply:
Yeah it ain't pretty, not terribly future proof (this very PR is evidence thereof). 

I believe we'd need to dive into each exception in the list and fix the root causes instead of hacking it into this directive. Definitely a task worth looking at, but probably not one that will be done "just like that". 

I guess what I'm trying to say is I consider this more a patch for something broken than a permanent solution.

Reply:
So I think one step closer to a working solution is good.

But since your asking I guess I should share my view, I do agree that it seems like a hacky solution, not the fix, but the solution in general.

My current thought is that it would be better if we just checked wether the click target isn't inside the given target aka. the dropdown element. Thereby the functionality actually does what it says, instead of begin customised to work with specifics.

Reply:
Agreed. But as I wrote above, we need to fix the root causes for those hacks before we can remove them... which points at fixing stuff in tours, overlays and then obviously the save options pop-up.

Reply:
I would be okay with this fix, and then creating an issue in regards of fixing the solution, in general, later.

But I'm not sure if there is any policy regarding that approach? @Shazwazza

Reply:
All good, i was just curious to hear people's thoughts ... in hopes there was maybe a 'quick win' :)